static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -301,47 +303,102 @@ class SpanTreeModel {
       }
     }
 
-    const {descendants} = (hideSpanTree ? [] : descendantsSource).reduce(
+    // Check if the descendants in this span have consecutive similar siblings, and group them
+    const groupedDescendants: SpanTreeModel[][] = [];
+    if (descendantsSource?.length >= MIN_SIBLING_GROUP_SIZE) {
+      let prevSpanModel = descendantsSource[0];
+      let currentGroup = [prevSpanModel];
+
+      for (let i = 1; i < descendantsSource.length; i++) {
+        const currSpanModel = descendantsSource[i];
+
+        // We want to group siblings only if they share the same op and description
+        if (
+          prevSpanModel.span.op === currSpanModel.span.op &&
+          prevSpanModel.span.description === currSpanModel.span.description
+        ) {
+          currentGroup.push(currSpanModel);
+        } else {
+          groupedDescendants.push(currentGroup);
+          currentGroup = [currSpanModel];
+        }
+
+        prevSpanModel = currSpanModel;
+      }
+
+      groupedDescendants.push(currentGroup);
+    } else if (descendantsSource.length >= 1) {
+      groupedDescendants.push(descendantsSource);
+    }
+
+    const {descendants} = (hideSpanTree ? [] : groupedDescendants).reduce(
       (
         acc: {
           descendants: EnhancedProcessedSpanType[];
           previousSiblingEndTimestamp: number | undefined;
         },
-        span,
+        group,
         index
       ) => {
-        acc.descendants.push(
-          ...span.getSpansList({
-            operationNameFilters,
-            generateBounds,
-            treeDepth: shouldHideSpanOfGroup ? treeDepth : treeDepth + 1,
-            isLastSibling: index === lastIndex,
-            continuingTreeDepths: descendantContinuingTreeDepths,
-            hiddenSpanGroups,
-            spanGroups: new Set(childSpanGroup),
-            filterSpans,
-            previousSiblingEndTimestamp: acc.previousSiblingEndTimestamp,
-            event,
-            isOnlySibling: descendantsSource.length === 1,
-            spanGrouping: shouldGroup
-              ? [...(spanGrouping ?? []), wrappedSpan]
-              : undefined,
-            toggleSpanGroup: isNotLastSpanOfGroup
-              ? toggleSpanGroup === undefined
-                ? this.toggleSpanGroup
-                : toggleSpanGroup
-              : undefined,
-            showSpanGroup: isNotLastSpanOfGroup
-              ? toggleSpanGroup === undefined
-                ? this.showSpanGroup
-                : showSpanGroup
-              : false,
-            addTraceBounds,
-            removeTraceBounds,
-          })
-        );
-
-        acc.previousSiblingEndTimestamp = span.span.timestamp;
+        // Groups less than 5 indicate that the spans should be left ungrouped
```

Comment:
@0Calories I think there's a simpler implementation where we can avoid adding anything to the `reduce()` callback here. 

You can create an observable mobx dictionary, maybe called `siblingGroup`, as a top-level field on `SpanTreeModel` with the following types:

```
type Occurrence = number;

// determines if the Nth occurance of the (SpanOperationName, SpanDescription) sibling group should be expanded. Otherwise, auto-group them.
type OccuranceDisplayMap = Record<Occurrence, ShouldDisplay>

SpanTreeModel.siblingGroup = Record<SpanOperationName, Record<SpanDescription, OccuranceDisplayMap>>
```

This should control the toggling of the sibling group when filtering on `descendantsSource`. Similar to `hiddenSpanGroups` here: https://github.com/getsentry/sentry/blob/b498737932532269c2d98a40fcda8c5febce5deb/static/app/components/events/interfaces/spans/waterfallModel.tsx#L33

I can walk you thru this proposal if there's any confusion.